src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -400,9 +409,14 @@ public synchronized void execute(BuildResult buildResult) {
     }
 
     private void captureTestParallelization() {
-        gradle.afterProject(p ->
-            p.getTasks().withType(Test.class).configureEach(captureMaxParallelForks(buildScan))
-        );
+        gradle.afterProject(p -> {
+            TaskCollection<Test> tests = p.getTasks().withType(Test.class);
+            if (GradleVersion.current().compareTo(GradleVersion.version("5.0")) >= 0) {
+                tests.configureEach(captureMaxParallelForks(buildScan));
+            } else {
+                tests.all(captureMaxParallelForks(buildScan));
```

Comment:
I have no idea what the state of Lazy Configuration was in 4.x, so `all` may not be best here.

Reply:
LGTM